CheckWhoIsTheWinnerTask.java

```diff
@@ -1,20 +1,27 @@
 public class CheckWhoIsTheWinnerTask {
     public static void main(String[] args) {
+        //write the situation using char matrix case.
         char [][] mainMatrix = {
                 {'X', 'O', 'X'},
                 {'X', 'X', 'O'},
                 {'O', 'X', 'O'}
         };
-        boolean bInput = xOrO(mainMatrix);
-        if(bInput){
+        //saved the method in main booleans and then print them.
+        boolean bInputPlayerX = xOrO(mainMatrix, 'X');
+        boolean bInputPlayerO = xOrO(mainMatrix, 'O');
+        if(bInputPlayerX){
+            //printed X player condition of winning.
             System.out.println("X player has won the game.");
-        }else if(!bInput){
+        }else if(bInputPlayerO){
+            //printed O player condition of winning.
             System.out.println("O player has won the game.");
         }else{
+            //printed the draw in else
             System.out.println("It is a draw, no one won the game.");
         }
     }
-    public static boolean xOrO (char [][] matrix){
+    //returned the condition to get the win or a draw.
+    public static boolean xOrO(char[][] matrix, char ignoredPlayer){

```

Comment:
what is ognoredplayer? you get argument x you need to check x you get o you need to check 0

Reply:
always check your code before sending also

Reply:
player is x when you call xOrO(mainMatrix, 'X');
and player is o when you call xOrO(mainMatrix, '0');